### PR TITLE
feature: WEOS-1382 As a developer I should be able to set the type of id that I want auto generated

### PR DESCRIPTION
**Pull request by shaniah868**
* **Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

-Now you can auto generate your x-identifier once only one is specified based on its format
-If the format is not specified and error is thrown

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

**Review by Aidan-E — DISMISSED** [review]

File: model/content_entity.go [hunk 1]
```diff
@@ -749,3 +751,44 @@ func (w *ContentEntity) UnmarshalJSON(data []byte) error {
 	err = json.Unmarshal(data, &w.Property)
 	return err
 }
+
+//GenerateID adds a generated id to the payload based on the schema
+func GenerateID(payload []byte, entityFactory EntityFactory) ([]byte, error) {
+	schema := entityFactory.Schema()
```
Comment (akeemphilbert) [inline model/content_entity.go]: why are you passing in the EntityFactory? the schema is already on the object (FromSchemaAndBuilder)

File: model/content_entity.go [hunk 2]
```diff
@@ -749,3 +751,44 @@ func (w *ContentEntity) UnmarshalJSON(data []byte) error {
...
+	properties := entityFactory.Schema().ExtensionProps.Extensions["x-identifier"]
```
Comment (akeemphilbert) [inline model/content_entity.go]: though passing in the schema is not needed, you set a variable "schema" then you still use the schema on the entity factory?

File: model/content_entity.go [hunk 3]
```diff
@@ -749,3 +751,44 @@ func (w *ContentEntity) UnmarshalJSON(data []byte) error {
 	err = json.Unmarshal(data, &w.Property)
 	return err
 }
+
+//GenerateID adds a generated id to the payload based on the schema
+func GenerateID(payload []byte, entityFactory EntityFactory) ([]byte, error) {
```
Comment (akeemphilbert) [inline model/content_entity.go]: This should be a function ON a content entity that is maybe triggered on an isValid call. The isvalid could check to see if the identifier fields have value and if the identifier is one field then trigger the generateID function

File: model/receiver.go [hunk 4]
```diff
@@ -36,7 +36,12 @@ func CreateHandler(ctx context.Context, command *Command, eventStore EventReposi
 		logger.Debug(err)
 		return err
 	}
-
+	//Generate id for entity
+	payload, err = GenerateID(payload, entityFactory)
```
Comment (akeemphilbert) [inline model/receiver.go]: this should have been instantiating the ContentEntity with the payload (maybe we need a construction FromSchemaAndBuilderAndPayload)

**Review by akeemphilbert — CHANGES_REQUESTED** [review]
When I removed "And" from one of the steps so that is uses the same "the specification is" as the other features, the tests failed

**Review by akeemphilbert — APPROVED** [review]

**Review by Aidan-E — APPROVED** [review]

**Pull request merged**